DataScience/AisAudioLabeler.py

```diff
@@ -0,0 +1,371 @@
+                        os.mkdir(copy_path)
+                    # TODO: Remove str() once using Python 3.9+
+                    shutil.move(str(download_path / name), str(copy_path / name))
+
+
+def load_ais_files(inp_path):
+    """Load all AIS files residing on the input path containing
+    required keys.
+
+    Note that AIS files contain a single AIS sample using JSON on each
+    line.
+
+    Parameters
+    ----------
+    inp_path : pathlib.Path()
+        Path to directory containing JSON files
+
+    Returns
+    -------
+    ais : pd.DataFrame()
+        AIS position samples with ship type
+
+    """
+    positions = []
+    position_keys = set(["mmsi", "status", "timestamp", "speed", "lat", "lon"])
+    types = {}
+    type_keys = set(["mmsi", "shiptype"])
+    names = os.listdir(inp_path)
+    n_lines = 0
+    n_positions = 0
+    n_types = 0
+    for name in names:
+        with open(inp_path / name, "r") as f:
+            for line in f:
+                n_lines += 1
+                sample = json.loads(line)
+                if position_keys.issubset(set(sample.keys())):
+                    # Collect samples containing position
+                    n_positions += 1
+                    positions.append(sample)
+                elif type_keys.issubset(set(sample.keys())):
+                    # Collect samples containing the mapping from mmsi to shiptype
+                    n_types += 1
+                    if sample["mmsi"] not in types:
+                        types[sample["mmsi"]] = sample["shiptype"]
+
+    ais = pd.DataFrame(positions).sort_values(by=["timestamp"], ignore_index=True)
+    ais["shiptype"] = ais["mmsi"].apply(lambda x: types[x] if x in types else None)
+    logger.info(f"Read {n_lines} lines")
+    logger.info(f"Found {n_positions} positions")
+    logger.info(f"Found {n_types} types")
+    return ais
+
+
+def get_hydrophone_metadata(inp_path):
+    """Probe all audio files residing on the input path.
+
+    Parameters
+    ----------
+    inp_path : pathlib.Path()
+        Path to directory containing audio files
+
+    Returns
+    -------
+    hmd : pd.DataFrame()
+        Audio stream entries
+
+    """
+    entries = []
+    req_keys = set(["sample_rate", "duration"])
+    names = os.listdir(inp_path)
+    for name in names:
+        entry = lu.probe_audio_file(inp_path / name)
+        if not req_keys.issubset(set(entry.keys())):
+            continue
+        entry["sample_rate"] = int(entry["sample_rate"])
+        entry["duration"] = float(entry["duration"])
+        entry["name"] = name
+        s = re.search(r"-([0-9]+)-[a-zA-Z]+\.", name)
+        if s is not None:
+            start_timestamp = s.group(1)
+        else:
+            start_timestamp = s.group(1)
+        entry["start_timestamp"] = int(start_timestamp)
+        entries.append(entry)
+    hmd = pd.DataFrame(entries).sort_values(by=["start_timestamp"], ignore_index=True)
+    return hmd
+
+
+def main():
```

Comment:
This function has a lot going on in it. Should the graphing stuff be split into a different function?

Reply:
Yes, I agree ... will do on the next branch.